packages/page/src/Application/MessageHandler/CopyPageMessageHandler.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Page\Application\MessageHandler;
+
+use Sulu\Component\Localization\Manager\LocalizationManagerInterface;
+use Sulu\Content\Application\ContentCopier\ContentCopierInterface;
+use Sulu\Content\Domain\Model\DimensionContentInterface;
+use Sulu\Page\Application\Message\CopyPageMessage;
+use Sulu\Page\Domain\Repository\PageRepositoryInterface;
+
+/**
+ * @experimental
+ *
+ * @internal This class should not be instantiated by a project.
+ *           Create your own Message and Handler instead.
+ */
+final class CopyPageMessageHandler
+{
+    public function __construct(
+        private PageRepositoryInterface $pageRepository,
+        private ContentCopierInterface $contentCopier,
+        private LocalizationManagerInterface $localizationManager,
+    ) {
+    }
+
+    public function __invoke(CopyPageMessage $message): void
+    {
+        $page = $this->pageRepository->getOneBy($message->getIdentifier());
```

Comment:
Would use speaking variables:

```suggestion
        $sourcePage = $this->pageRepository->getOneBy($message->getIdentifier());
```